### PR TITLE
PEP8 cleanup - ask.py

### PR DESCRIPTION
**Pull request by jremes-foss**
Lines 84 and 145 had some PEP8 issues, as the use of "==" and "is" comparisons were inconsistent. These have been amended. There was also some under-intended lines and long lines, but I am not sure what is the official line of these - are the lines allowed to be over 79 characters? If not, I can start amending these on separate PR. Thanks.

### CODE REVIEW

**Comment by moorepants** [comment]
If the tests pass, +1.

File: sympy/assumptions/ask.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def ask(proposition, assumptions=True, context=global_assumptions):
     if res is not None:
         return bool(res)
 
-    if assumptions == True:
+    if assumptions is True:
```
Comment (toolforger) [inline sympy/assumptions/ask.py]: Why not just ```if assumptions:``` ?

Also, ```assumptions``` is the result of a ```to_cnf``` call, so it might be an unevaluated ```And``` object, which certainly is not going to be ```is True```.
I see all tests pass, but maybe that's just because the code after this test comes to the same conclusion, while just taking some extra hoops. (Or maybe the ```is``` test would be faster, if the code inside this function is faster than the ``bool()``` member of ```And```.)
tl;dr: Just looking at the tests is not enough, unless the tests specifically cover the case that the CNF is an ```And```ed compound.

**Comment by Newman101** [comment]
if assumptions: is fair enough as well, as far as I know that is PEP8 compatible.

**Comment by debugger22** [comment]
> if assumptions: is fair enough as well, as far as I know that is PEP8 compatible

I'm not sure if it is PEP8 compatible but this syntax is very heavily used in SymPy.

**Comment by moorepants** [comment]
`if x:` isn't a "style" issue, thus PEP8 is irrelevant. These things may seem similar but in fact function differently:

```
if x:
if x is True:
if x == True:
```

**Comment by moorepants** [comment]
PEP8 does mention this:

https://www.python.org/dev/peps/pep-0008/#programming-recommendations

but it is only with respect to boolean values:

```
Don't compare boolean values to True or False using == .

Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```

**Comment by debugger22** [comment]
> but it is only with respect to boolean values:

We do not expect anything other than boolean from assumptions except `None`. So I think it makes sense to have just  `if expr.is_*`.

**Comment by toolforger** [comment]
@Newman101 can you check that there is at least one unit test that covers the case that line 145 gets hit by an ```And``` object?
If no, I'd want to such a unit test added.

**Comment by toolforger** [comment]
Sweet, thanks.

**Comment by Newman101** [comment]
Found one from test_query.py.

assert ask(Q.bounded(a), Q.bounded(x) & Q.bounded(y)) is True

ask syntax is ask(proposition, assumptions), so this gets hit by And object in the unit tests.

File: sympy/assumptions/ask.py (hunk 1 above)
Comment (certik) [inline sympy/assumptions/ask.py]: @Newman101 do you want to do this change?

**Comment by certik** [comment]
@Newman101 let us know if you want to do the change that @toolforger suggested above.

**Comment by czgdp1807** [comment]
The changes have been made or the lines to be fixed are removed from master. Therefore closing this PR. Please let us know if there is a need to re-open this. Thanks for the contributions.
